Keep compatibility experimental file_name RPCs for now.

Index: hurd-debian/exec/hashexec.c
===================================================================
--- hurd-debian.orig/exec/hashexec.c
+++ hurd-debian/exec/hashexec.c
@@ -24,6 +24,9 @@
 #include <unistd.h>
 #include <envz.h>
 #include <sys/param.h>
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+#include <hurd/fs_experimental.h>
+#endif
 
 /* This is called to check E for a #! interpreter specification.  E has
    already been prepared (successfully) and checked (unsuccessfully).  If
@@ -447,6 +450,21 @@ check_hashbang (struct execdata *e,
   /* For backwards compatibility.  Just drop it when we kill file_exec.  */
   if (e->error == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+  /* Execute the interpreter program.  */
+  e->error = file_exec_file_name (interp_file,
+				  oldtask, flags, interp,
+				  new_argv, new_argvlen, envp, envplen,
+				  new_dtable ?: dtable,
+				  MACH_MSG_TYPE_COPY_SEND,
+				  new_dtable ? new_dtablesize : dtablesize,
+				  portarray, MACH_MSG_TYPE_COPY_SEND, nports,
+				  intarray, nints,
+				  deallocnames, ndeallocnames,
+				  destroynames, ndestroynames);
+  /* For backwards compatibility.  Just drop it when we kill file_exec.  */
+  if (e->error == MIG_BAD_ID)
+#endif
     e->error = file_exec (interp_file,
 			  oldtask, flags,
 			  new_argv, new_argvlen, envp, envplen,
Index: hurd-debian/startup/startup.c
===================================================================
--- hurd-debian.orig/startup/startup.c
+++ hurd-debian/startup/startup.c
@@ -24,6 +24,9 @@
    one file. */
 #include <hurd.h>
 #include <hurd/fs.h>
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+#include <hurd/fs_experimental.h>
+#endif
 #include <hurd/fsys.h>
 #include <device/device.h>
 #include <stdio.h>
@@ -417,6 +420,21 @@ run (const char *server, mach_port_t *po
 	     file_exec.  */
 	  if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+	  err = file_exec_file_name (file, *task, 0, (char *)prog,
+				     argz,
+				     argz_len, /* Args.  */
+				     startup_envz, startup_envz_len,
+				     default_dtable,
+				     MACH_MSG_TYPE_COPY_SEND, 3,
+				     ports, MACH_MSG_TYPE_COPY_SEND,
+				     INIT_PORT_MAX,
+				     default_ints, INIT_INT_MAX,
+				     NULL, 0, NULL, 0);
+	  /* For backwards compatibility.  Just drop it when we kill
+	     file_exec.  */
+	  if (err == MIG_BAD_ID)
+#endif
 	    err = file_exec (file, *task, 0,
 			     argz, argz_len, /* Args.  */
 			     startup_envz, startup_envz_len,
@@ -565,6 +583,18 @@ run_for_real (char *filename, char *args
   /* For backwards compatibility.  Just drop it when we kill file_exec.  */
   if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+  err = file_exec_file_name (file, task, 0, filename,
+			     args, arglen,
+			     startup_envz, startup_envz_len,
+			     default_dtable, MACH_MSG_TYPE_COPY_SEND, 3,
+			     default_ports, MACH_MSG_TYPE_COPY_SEND,
+			     INIT_PORT_MAX,
+			     default_ints, INIT_INT_MAX,
+			     NULL, 0, NULL, 0);
+  /* For backwards compatibility.  Just drop it when we kill file_exec.  */
+  if (err == MIG_BAD_ID)
+#endif
     err = file_exec (file, task, 0,
 		     args, arglen,
 		     startup_envz, startup_envz_len,
@@ -1284,6 +1314,18 @@ start_child (const char *prog, char **pr
   /* For backwards compatibility.  Just drop it when we kill file_exec.  */
   if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+  err = file_exec_file_name (file, child_task, 0, args,
+			     args, arglen,
+			     startup_envz, startup_envz_len,
+			     NULL, MACH_MSG_TYPE_COPY_SEND, 0, /* No fds.  */
+			     default_ports, MACH_MSG_TYPE_COPY_SEND,
+			     INIT_PORT_MAX,
+			     default_ints, INIT_INT_MAX,
+			     NULL, 0, NULL, 0);
+  /* For backwards compatibility.  Just drop it when we kill file_exec.  */
+  if (err == MIG_BAD_ID)
+#endif
     err = file_exec (file, child_task, 0,
 		     args, arglen,
 		     startup_envz, startup_envz_len,
Index: hurd-debian/libdiskfs/file-exec.c
===================================================================
--- hurd-debian.orig/libdiskfs/file-exec.c
+++ hurd-debian/libdiskfs/file-exec.c
@@ -22,10 +22,14 @@
 
 #include "priv.h"
 #include "fs_S.h"
+#include "fs_experimental_S.h"
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <hurd/exec.h>
 #include <hurd/paths.h>
+#ifdef HAVE_EXEC_EXEC_FILE_NAME
+#include <hurd/exec_experimental.h>
+#endif
 #include <string.h>
 #include <idvec.h>
 
@@ -63,6 +67,40 @@ diskfs_S_file_exec (struct protid *cred,
 }
 
 kern_return_t
+diskfs_S_file_exec_file_name (struct protid *cred,
+		    task_t task,
+		    int flags,
+		    char *filename,
+		    char *argv,
+		    size_t argvlen,
+		    char *envp,
+		    size_t envplen,
+		    mach_port_t *fds,
+		    size_t fdslen,
+		    mach_port_t *portarray,
+		    size_t portarraylen,
+		    int *intarray,
+		    size_t intarraylen,
+		    mach_port_t *deallocnames,
+		    size_t deallocnameslen,
+		    mach_port_t *destroynames,
+		    size_t destroynameslen)
+{
+  return diskfs_S_file_exec_paths (cred,
+				       task,
+				       flags,
+				       filename,
+				       filename,
+				       argv, argvlen,
+				       envp, envplen,
+				       fds, fdslen,
+				       portarray, portarraylen,
+				       intarray, intarraylen,
+				       deallocnames, deallocnameslen,
+				       destroynames, destroynameslen);
+}
+
+kern_return_t
 diskfs_S_file_exec_paths (struct protid *cred,
 			  task_t task,
 			  int flags,
@@ -209,6 +247,21 @@ diskfs_S_file_exec_paths (struct protid
 	  /* For backwards compatibility.  Just drop it when we kill
 	     exec_exec.  */
 	  if (err == MIG_BAD_ID)
+#endif
+#ifdef HAVE_EXEC_EXEC_FILE_NAME
+	  err = exec_exec_file_name (execserver,
+				     right, MACH_MSG_TYPE_COPY_SEND,
+				     task, flags, path,
+				     argv, argvlen, envp, envplen,
+				     fds, MACH_MSG_TYPE_COPY_SEND, fdslen,
+				     portarray, MACH_MSG_TYPE_COPY_SEND,
+				     portarraylen,
+				     intarray, intarraylen,
+				     deallocnames, deallocnameslen,
+				     destroynames, destroynameslen);
+	  /* For backwards compatibility.  Just drop it when we kill
+	     exec_exec.  */
+	  if (err == MIG_BAD_ID)
 #endif
 	    err = exec_exec (execserver,
 			     right, MACH_MSG_TYPE_COPY_SEND,
Index: hurd-debian/libfshelp/start-translator-long.c
===================================================================
--- hurd-debian.orig/libfshelp/start-translator-long.c
+++ hurd-debian/libfshelp/start-translator-long.c
@@ -29,6 +29,9 @@
 #include <string.h>
 #include <assert-backtrace.h>
 #include "fshelp.h"
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+#include <hurd/fs_experimental.h>
+#endif
 
 
 /* The data passed in the various messages we're interested in.  */
@@ -284,6 +287,16 @@ fshelp_start_translator_long (fshelp_ope
   /* For backwards compatibility.  Just drop it when we kill file_exec.  */
   if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+  /* Try and exec the translator in TASK...  */
+  err = file_exec_file_name (executable, task, EXEC_DEFAULTS, name,
+			     argz, argz_len, 0, 0,
+			     fds, fds_type, fds_len,
+			     ports, ports_type, ports_len,
+			     ints, ints_len, 0, 0, 0, 0);
+  /* For backwards compatibility.  Just drop it when we kill file_exec.  */
+  if (err == MIG_BAD_ID)
+#endif
     err = file_exec (executable, task, EXEC_DEFAULTS,
 		     argz, argz_len, 0, 0,
 		     fds, fds_type, fds_len,
Index: hurd-debian/libnetfs/file-exec.c
===================================================================
--- hurd-debian.orig/libnetfs/file-exec.c
+++ hurd-debian/libnetfs/file-exec.c
@@ -24,10 +24,14 @@
 #include "netfs.h"
 #include "execserver.h"
 #include "fs_S.h"
+#include "fs_experimental_S.h"
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <hurd/exec.h>
 #include <hurd/paths.h>
+#ifdef HAVE_EXEC_EXEC_FILE_NAME
+#include <hurd/exec_experimental.h>
+#endif
 #include <string.h>
 #include <idvec.h>
 
@@ -65,6 +69,40 @@ netfs_S_file_exec (struct protid *cred,
 }
 
 kern_return_t
+netfs_S_file_exec_file_name (struct protid *cred,
+		   task_t task,
+		   int flags,
+		   char *filename,
+		   char *argv,
+		   size_t argvlen,
+		   char *envp,
+		   size_t envplen,
+		   mach_port_t *fds,
+		   size_t fdslen,
+		   mach_port_t *portarray,
+		   size_t portarraylen,
+		   int *intarray,
+		   size_t intarraylen,
+		   mach_port_t *deallocnames,
+		   size_t deallocnameslen,
+		   mach_port_t *destroynames,
+		   size_t destroynameslen)
+{
+  return netfs_S_file_exec_paths (cred,
+				      task,
+				      flags,
+				      filename,
+				      filename,
+				      argv, argvlen,
+				      envp, envplen,
+				      fds, fdslen,
+				      portarray, portarraylen,
+				      intarray, intarraylen,
+				      deallocnames, deallocnameslen,
+				      destroynames, destroynameslen);
+}
+
+kern_return_t
 netfs_S_file_exec_paths (struct protid *cred,
 			 task_t task,
 			 int flags,
@@ -183,6 +221,21 @@ netfs_S_file_exec_paths (struct protid *
 	      /* For backwards compatibility.  Just drop it when we kill
 		 exec_exec.  */
 	      if (err == MIG_BAD_ID)
+#endif
+#ifdef HAVE_EXEC_EXEC_FILE_NAME
+	      err = exec_exec_file_name (_netfs_exec,
+					 right, MACH_MSG_TYPE_COPY_SEND,
+					 task, flags, path,
+					 argv, argvlen, envp, envplen,
+					 fds, MACH_MSG_TYPE_COPY_SEND, fdslen,
+					 portarray, MACH_MSG_TYPE_COPY_SEND,
+					 portarraylen,
+					 intarray, intarraylen,
+					 deallocnames, deallocnameslen,
+					 destroynames, destroynameslen);
+	      /* For backwards compatibility.  Just drop it when we kill
+		 exec_exec.  */
+	      if (err == MIG_BAD_ID)
 #endif
 		err = exec_exec (_netfs_exec,
 				 right, MACH_MSG_TYPE_COPY_SEND,
Index: hurd-debian/libtrivfs/file-exec.c
===================================================================
--- hurd-debian.orig/libtrivfs/file-exec.c
+++ hurd-debian/libtrivfs/file-exec.c
@@ -43,6 +43,31 @@ trivfs_S_file_exec (trivfs_protid_t exec
 }
 
 kern_return_t
+trivfs_S_file_exec_file_name (trivfs_protid_t exec_file,
+			      mach_port_t reply,
+			      mach_msg_type_name_t replyPoly,
+			      mach_port_t exec_task,
+			      int flags,
+			      string_t filename,
+			      data_t argv,
+			      mach_msg_type_number_t argvCnt,
+			      data_t envp,
+			      mach_msg_type_number_t envpCnt,
+			      portarray_t fdarray,
+			      mach_msg_type_number_t fdarrayCnt,
+			      portarray_t portarray,
+			      mach_msg_type_number_t portarrayCnt,
+			      intarray_t intarray,
+			      mach_msg_type_number_t intarrayCnt,
+			      mach_port_array_t deallocnames,
+			      mach_msg_type_number_t deallocnamesCnt,
+			      mach_port_array_t destroynames,
+			      mach_msg_type_number_t destroynamesCnt)
+{
+  return EOPNOTSUPP;
+}
+
+kern_return_t
 trivfs_S_file_exec_paths (trivfs_protid_t exec_file,
 			  mach_port_t reply,
 			  mach_msg_type_name_t replyPoly,
Index: hurd-debian/trans/fakeroot.c
===================================================================
--- hurd-debian.orig/trans/fakeroot.c
+++ hurd-debian/trans/fakeroot.c
@@ -28,10 +28,14 @@
 #include <sys/sysmacros.h>
 #include <hurd/ihash.h>
 #include <hurd/paths.h>
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+#include <hurd/fs_experimental.h>
+#endif
 
 #include <version.h>
 
 #include "libnetfs/fs_S.h"
+#include "libnetfs/fs_experimental_S.h"
 #include "libnetfs/io_S.h"
 #include "libnetfs/fsys_S.h"
 #include "libports/notify_S.h"
@@ -937,6 +941,24 @@ netfs_S_file_exec_paths (struct protid *
 	 file_exec.  */
       if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+      /* We cannot use MACH_MSG_TYPE_MOVE_SEND because we might need to
+	 retry an interrupted call that would have consumed the rights.  */
+      err = file_exec_file_name (netfs_node_netnode (user->po->np)->file,
+				 task, flags,
+				 path,
+				 argv, argvlen,
+				 envp, envplen,
+				 fds, MACH_MSG_TYPE_COPY_SEND, fdslen,
+				 portarray, MACH_MSG_TYPE_COPY_SEND,
+				 portarraylen,
+				 intarray, intarraylen,
+				 deallocnames, deallocnameslen,
+				 destroynames, destroynameslen);
+      /* For backwards compatibility.  Just drop it when we kill
+	 file_exec.  */
+      if (err == MIG_BAD_ID)
+#endif
 	err = file_exec (user->po->np->nn->file, task, flags, argv, argvlen,
 			 envp, envplen, fds, MACH_MSG_TYPE_COPY_SEND, fdslen,
 			 portarray, MACH_MSG_TYPE_COPY_SEND, portarraylen,
@@ -991,6 +1013,40 @@ netfs_S_file_exec (struct protid *user,
 				  destroynames, destroynameslen);
 }
 
+kern_return_t
+netfs_S_file_exec_file_name (struct protid *user,
+                   task_t task,
+                   int flags,
+                   char *filename,
+                   char *argv,
+                   size_t argvlen,
+                   char *envp,
+                   size_t envplen,
+                   mach_port_t *fds,
+                   size_t fdslen,
+                   mach_port_t *portarray,
+                   size_t portarraylen,
+                   int *intarray,
+                   size_t intarraylen,
+                   mach_port_t *deallocnames,
+                   size_t deallocnameslen,
+                   mach_port_t *destroynames,
+                   size_t destroynameslen)
+{
+  return netfs_S_file_exec_paths (user,
+				      task,
+				      flags,
+				      filename,
+				      filename,
+				      argv, argvlen,
+				      envp, envplen,
+				      fds, fdslen,
+				      portarray, portarraylen,
+				      intarray, intarraylen,
+				      deallocnames, deallocnameslen,
+				      destroynames, destroynameslen);
+}
+
 error_t
 netfs_S_io_map (struct protid *user,
 		mach_port_t *rdobj, mach_msg_type_name_t *rdobjtype,
@@ -1107,6 +1163,7 @@ netfs_demuxer (mach_msg_header_t *inp,
   mig_routine_t routine;
   if ((routine = netfs_io_server_routine (inp)) ||
       (routine = netfs_fs_server_routine (inp)) ||
+      (routine = netfs_fs_experimental_server_routine (inp)) ||
       (routine = ports_notify_server_routine (inp)) ||
       (routine = netfs_fsys_server_routine (inp)) ||
       /* XXX we should intercept interrupt_operation and do
Index: hurd-debian/utils/login.c
===================================================================
--- hurd-debian.orig/utils/login.c
+++ hurd-debian/utils/login.c
@@ -47,6 +47,9 @@
 #include <error.h>
 #include <timefmt.h>
 #include <hurd/lookup.h>
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+#include <hurd/fs_experimental.h>
+#endif
 #include <ugids.h>
 
 const char *argp_program_version = STANDARD_HURD_VERSION (login);
@@ -893,6 +896,16 @@ main(int argc, char *argv[])
   /* Fallback in case the file server hasn't been restarted.  */
   if (err == MIG_BAD_ID)
 #endif
+#ifdef HAVE_FILE_EXEC_FILE_NAME
+  err = file_exec_file_name (exec, mach_task_self (), EXEC_DEFAULTS, shell,
+			     sh_args, sh_args_len, env, env_len,
+			     fds, MACH_MSG_TYPE_COPY_SEND, 3,
+			     ports, MACH_MSG_TYPE_COPY_SEND, INIT_PORT_MAX,
+			     ints, INIT_INT_MAX,
+			     0, 0, 0, 0);
+  /* Fallback in case the file server hasn't been restarted.  */
+  if (err == MIG_BAD_ID)
+#endif
     err = file_exec (exec, mach_task_self (), EXEC_DEFAULTS,
 		     sh_args, sh_args_len, env, env_len,
 		     fds, MACH_MSG_TYPE_COPY_SEND, 3,
Index: hurd-debian/hurd/fs_experimental.defs
===================================================================
--- /dev/null
+++ hurd-debian/hurd/fs_experimental.defs
@@ -0,0 +1,51 @@
+/* Definitions for the filesystem interface.
+   Copyright (C) 1994,95,96,97,98,99,2002 Free Software Foundation, Inc.
+
+This file is part of the GNU Hurd.
+
+The GNU Hurd is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2, or (at your option)
+any later version.
+
+The GNU Hurd is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with the GNU Hurd; see the file COPYING.  If not, write to
+the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.  */
+
+
+/* All these objects also implement the generic IO facilities. */
+
+subsystem fs_experimental 444242;
+
+#include <hurd/hurd_types.defs>
+
+#ifdef FILE_IMPORTS
+FILE_IMPORTS
+#endif
+
+/* Operations supported on all files */
+
+INTR_INTERFACE
+
+/* Overlay a task with a file.  Necessary initialization, including
+   authentication changes associated with set[ug]id execution must be
+   handled by the filesystem.  Filesystems normally implement this by
+   using exec_newtask or exec_loadtask as appropriate.  */
+routine file_exec_file_name (
+	exec_file: file_t;
+	RPT
+	exec_task: task_t;
+	flags: int;
+	filename: string_t;
+	argv: data_t SCP;
+	envp: data_t SCP;
+	fdarray: portarray_t SCP;
+	portarray: portarray_t SCP;
+	intarray: intarray_t SCP;
+	deallocnames: mach_port_name_array_t SCP;
+	destroynames: mach_port_name_array_t SCP);
Index: hurd-debian/libdiskfs/Makefile
===================================================================
--- hurd-debian.orig/libdiskfs/Makefile
+++ hurd-debian/libdiskfs/Makefile
@@ -56,7 +56,7 @@ OTHERSRCS = conch-fetch.c conch-set.c di
 SRCS = $(OTHERSRCS) $(FSSRCS) $(IOSRCS) $(FSYSSRCS) $(IFSOCKSRCS)
 installhdrs = diskfs.h diskfs-pager.h
 
-MIGSTUBS = fsServer.o ioServer.o fsysServer.o exec_startupServer.o \
+MIGSTUBS = fsServer.o fs_experimentalServer.o ioServer.o fsysServer.o exec_startupServer.o \
 	fsys_replyUser.o fs_notifyUser.o ifsockServer.o \
 	startup_notifyServer.o
 OBJS = $(sort $(SRCS:.c=.o) $(MIGSTUBS))
@@ -66,6 +66,7 @@ LDLIBS += -lpthread
 
 fsys-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h -DREPLY_PORTS
 fs-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h
+fs_experimental-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h
 io-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h
 ifsock-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h
 exec_startup-MIGSFLAGS = -imacros $(srcdir)/fsmutations.h
Index: hurd-debian/libdiskfs/demuxer.c
===================================================================
--- hurd-debian.orig/libdiskfs/demuxer.c
+++ hurd-debian/libdiskfs/demuxer.c
@@ -19,6 +19,7 @@
 
 #include "io_S.h"
 #include "fs_S.h"
+#include "fs_experimental_S.h"
 #include "../libports/notify_S.h"
 #include "fsys_S.h"
 #include "../libports/interrupt_S.h"
@@ -33,6 +34,7 @@ diskfs_demuxer (mach_msg_header_t *inp,
   mig_routine_t routine;
   if ((routine = diskfs_io_server_routine (inp)) ||
       (routine = diskfs_fs_server_routine (inp)) ||
+      (routine = diskfs_fs_experimental_server_routine (inp)) ||
       (routine = ports_notify_server_routine (inp)) ||
       (routine = diskfs_fsys_server_routine (inp)) ||
       (routine = ports_interrupt_server_routine (inp)) ||
Index: hurd-debian/libnetfs/Makefile
===================================================================
--- hurd-debian.orig/libnetfs/Makefile
+++ hurd-debian/libnetfs/Makefile
@@ -59,12 +59,13 @@ SRCS= $(OTHERSRCS) $(FSSRCS) $(IOSRCS) $
 
 installhdrs=netfs.h
 
-MIGSTUBS= ioServer.o fsServer.o fsysServer.o fsys_replyUser.o ifsockServer.o
+MIGSTUBS= ioServer.o fsServer.o fs_experimentalServer.o fsysServer.o fsys_replyUser.o ifsockServer.o
 
 OBJS=$(sort $(SRCS:.c=.o) $(MIGSTUBS))
 
 fsys-MIGSFLAGS = -imacros $(srcdir)/mutations.h -DREPLY_PORTS
 fs-MIGSFLAGS = -imacros $(srcdir)/mutations.h
+fs_experimental-MIGSFLAGS = -imacros $(srcdir)/mutations.h
 io-MIGSFLAGS = -imacros $(srcdir)/mutations.h
 ifsock-MIGSFLAGS = -imacros $(srcdir)/mutations.h
 MIGCOMSFLAGS = -prefix netfs_
Index: hurd-debian/libnetfs/demuxer.c
===================================================================
--- hurd-debian.orig/libnetfs/demuxer.c
+++ hurd-debian/libnetfs/demuxer.c
@@ -22,6 +22,7 @@
 
 #include "io_S.h"
 #include "fs_S.h"
+#include "fs_experimental_S.h"
 #include "../libports/notify_S.h"
 #include "fsys_S.h"
 #include "../libports/interrupt_S.h"
@@ -34,6 +35,7 @@ netfs_demuxer (mach_msg_header_t *inp,
   mig_routine_t routine;
   if ((routine = netfs_io_server_routine (inp)) ||
       (routine = netfs_fs_server_routine (inp)) ||
+      (routine = netfs_fs_experimental_server_routine (inp)) ||
       (routine = ports_notify_server_routine (inp)) ||
       (routine = netfs_fsys_server_routine (inp)) ||
       (routine = ports_interrupt_server_routine (inp)) ||
Index: hurd-debian/libtrivfs/Makefile
===================================================================
--- hurd-debian.orig/libtrivfs/Makefile
+++ hurd-debian/libtrivfs/Makefile
@@ -44,7 +44,7 @@ OTHERSRCS=demuxer.c protid-clean.c proti
 
 SRCS=$(FSSRCS) $(IOSRCS) $(FSYSSRCS) $(OTHERSRCS)
 
-MIGSTUBS=fsServer.o ioServer.o fsysServer.o fsys_replyUser.o
+MIGSTUBS=fsServer.o fs_experimentalServer.o ioServer.o fsysServer.o fsys_replyUser.o
 
 libname = libtrivfs
 HURDLIBS = fshelp iohelp ports shouldbeinlibc
Index: hurd-debian/libtrivfs/demuxer.c
===================================================================
--- hurd-debian.orig/libtrivfs/demuxer.c
+++ hurd-debian/libtrivfs/demuxer.c
@@ -23,6 +23,7 @@
 
 #include "trivfs_io_S.h"
 #include "trivfs_fs_S.h"
+#include "trivfs_fs_experimental_S.h"
 #include "../libports/notify_S.h"
 #include "trivfs_fsys_S.h"
 #include "../libports/interrupt_S.h"
@@ -34,6 +35,7 @@ trivfs_demuxer (mach_msg_header_t *inp,
   mig_routine_t routine;
   if ((routine = trivfs_io_server_routine (inp)) ||
       (routine = trivfs_fs_server_routine (inp)) ||
+      (routine = trivfs_fs_experimental_server_routine (inp)) ||
       (routine = ports_notify_server_routine (inp)) ||
       (routine = trivfs_fsys_server_routine (inp)) ||
       (routine = ports_interrupt_server_routine (inp)))
Index: hurd-debian/configure.ac
===================================================================
--- hurd-debian.orig/configure.ac
+++ hurd-debian/configure.ac
@@ -163,6 +163,7 @@ else
     hurd_cv_libio=no))
 
 # Check if libc contains these functions.
+AC_CHECK_FUNCS(file_exec_file_name exec_exec_file_name)
 AC_CHECK_FUNCS(file_exec_paths exec_exec_paths _hurd_exec_paths)
 
 # Compatibility with glibc < 2.28
