Keep compatibility experimental file_name RPCs for now.

Index: hurd-debian/configure.ac
===================================================================
--- hurd-debian.orig/configure.ac
+++ hurd-debian/configure.ac
@@ -163,7 +163,7 @@ else
 fi
 AC_SUBST(VERSIONING)
 # Check if libc contains these functions.
-AC_CHECK_FUNCS(file_exec_file_name exec_exec_file_name)
+AC_CHECK_FUNCS(file_exec_file_name exec_exec_file_name _hurd_exec_file_name)
 AC_CHECK_FUNCS(file_exec_paths exec_exec_paths _hurd_exec_paths)
 
 
Index: hurd-debian/utils/fakeauth.c
===================================================================
--- hurd-debian.orig/utils/fakeauth.c
+++ hurd-debian/utils/fakeauth.c
@@ -422,6 +422,9 @@ believe it has restricted them to differ
 #ifdef HAVE__HURD_EXEC_PATHS
     err = _hurd_exec_paths (newtask, execfile, argv[argi], argv[argi],
 			    &argv[argi], environ);
+#elif defined HAVE__HURD_EXEC_FILE_NAME
+    err = _hurd_exec_file_name (newtask, execfile, argv[argi],
+				&argv[argi], environ);
 #else
     err = _hurd_exec (newtask, execfile, &argv[argi], environ);
 #endif
Index: hurd-debian/utils/rpctrace.c
===================================================================
--- hurd-debian.orig/utils/rpctrace.c
+++ hurd-debian/utils/rpctrace.c
@@ -1742,6 +1742,9 @@ traced_spawn (char **argv, char **envp)
 #ifdef HAVE__HURD_EXEC_PATHS
   err = _hurd_exec_paths (traced_task, file, prefixed_name ?: *argv,
 			  prefixed_name ?: *argv, argv, envp);
+#elif defined HAVE__HURD_EXEC_FILE_NAME
+  err = _hurd_exec_file_name (traced_task, file, prefixed_name ?: *argv,
+                              argv, envp);
 #else
   err = _hurd_exec (traced_task, file, argv, envp);
 #endif
Index: hurd-debian/utils/shd.c
===================================================================
--- hurd-debian.orig/utils/shd.c
+++ hurd-debian/utils/shd.c
@@ -159,6 +159,8 @@ run (char **argv, int fd0, int fd1)
 
 #ifdef HAVE__HURD_EXEC_PATHS
 	  err = _hurd_exec_paths (task, file, program, program, argv, environ);
+#elif defined HAVE__HURD_EXEC_FILE_NAME
+	  err = _hurd_exec_file_name (task, file, program, argv, environ);
 #else
 	  err = _hurd_exec (task, file, argv, environ);
 #endif
