Include DDE in the build

Index: hurd-debian/Makefile
===================================================================
--- hurd-debian.orig/Makefile
+++ hurd-debian/Makefile
@@ -33,6 +33,14 @@ lib-subdirs = libshouldbeinlibc libihash
 	      libbpf \
 	      libmachdev \
 
+ifneq ($(HAVE_LIBPCIACCESS),no)
+lib-subdirs += libmachdevdde libddekit
+endif
+
+ifeq ($(HAVE_PCAP),yes)
+lib-subdirs += eth-filter
+endif
+
 # Hurd programs
 prog-subdirs = auth proc exec term \
 	       ext2fs isofs tmpfs fatfs \
Index: hurd-debian/Makeconf
===================================================================
--- hurd-debian.orig/Makeconf
+++ hurd-debian/Makeconf
@@ -578,7 +578,7 @@ vpath %.defs $(top_srcdir)/hurd
 # These we want to find in the libc include directory...
 mach_defs_names = bootstrap exc mach mach4 \
 	mach_host mach_port mach_timer_reply memory_object \
-	memory_object_default notify \
+	memory_object_default notify experimental \
 	gnumach \
 	task_notify \
 
Index: hurd-debian/configure.ac
===================================================================
--- hurd-debian.orig/configure.ac
+++ hurd-debian/configure.ac
@@ -261,6 +261,8 @@ AC_SUBST(boot_store_types)dnl
 AC_MSG_CHECKING(boot store types)
 AC_MSG_RESULT($boot_store_types)
 
+AC_CHECK_HEADER([pcap.h], [HAVE_PCAP=yes], [HAVE_PCAP=no])
+
 # Check for ncursesw, which is needed for the console-curses client.
 hurd_LIB_NCURSESW
 
Index: hurd-debian/config.make.in
===================================================================
--- hurd-debian.orig/config.make.in
+++ hurd-debian/config.make.in
@@ -116,6 +116,9 @@ HAVE_LIBPCIACCESS = @HAVE_LIBPCIACCESS@
 libpciaccess_CFLAGS = @libpciaccess_CFLAGS@
 libpciaccess_LIBS = @libpciaccess_LIBS@
 
+# Whether pcap support is available.
+HAVE_PCAP = @HAVE_PCAP@
+
 # Installation tools.
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
