Index: hurd-debian/proc/wait.c
commit 909d32b5361eddda3a0db3586b1cb6598de7b620
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Mon Dec 28 23:25:29 2020 +0100

    proc: support waitid(WNOWAIT)
    
    * proc/wait.c (S_proc_waitid): When options contains WNOWAIT, do not set
    p_waited or complete child death.

===================================================================
--- hurd-debian.orig/proc/wait.c
+++ hurd-debian/proc/wait.c
@@ -189,13 +189,19 @@ S_proc_waitid (struct proc *p,
 		  || !(child->p_traced || (options & WUNTRACED)))
 	      && (!child->p_continued || !(options & WCONTINUED))))
 	return 0;
-      child->p_waited = 1;
+
       *status = child->p_status;
       *sigcode = child->p_sigcode;
       *ru = child->p_rusage; /* all zeros if !p_dead */
       *pid_status = child->p_pid;
-      if (child->p_dead)
-	complete_exit (child);
+
+      if (!(options & WNOWAIT))
+      {
+	child->p_waited = 1;
+	if (child->p_dead)
+	  complete_exit (child);
+      }
+
       return 1;
     }
 
