Just a build fix.

# missing vm_wire_all RPCs

To be removed with glibc 2.24-9

diff --git a/mach-defpager/Makefile b/mach-defpager/Makefile
index 3b8ce58e0..3cb506a27 100644
--- a/mach-defpager/Makefile
+++ b/mach-defpager/Makefile
@@ -23,11 +23,13 @@ dir	:= mach-defpager
 makemode:= server
 target	:= mach-defpager
 
+MIGSTUBS = gnumachUser.o
+
 SRCS	:= default_pager.c kalloc.c wiring.c main.c setup.c
 OBJS 	:= $(SRCS:.c=.o) \
 	   $(addsuffix Server.o,\
 		       memory_object default_pager memory_object_default exc) \
-	   default_pager_replyUser.o
+	   default_pager_replyUser.o $(MIGSTUBS)
 
 HURDLIBS:= ihash
 LDLIBS:= -lpthread
