Just a build fix.

# missing vm_wire_all RPCs

To be removed with glibc 2.24-9

Index: hurd-debian/mach-defpager/Makefile
===================================================================
--- hurd-debian.orig/mach-defpager/Makefile
+++ hurd-debian/mach-defpager/Makefile
@@ -23,11 +23,13 @@ dir	:= mach-defpager
 makemode:= server
 target	:= mach-defpager
 
+MIGSTUBS = gnumachUser.o
+
 SRCS	:= default_pager.c kalloc.c wiring.c main.c setup.c
 OBJS 	:= $(SRCS:.c=.o) \
 	   $(addsuffix Server.o,\
 		       memory_object default_pager memory_object_default exc) \
-	   default_pager_replyUser.o
+	   default_pager_replyUser.o $(MIGSTUBS)
 
 HURDLIBS:= ihash shouldbeinlibc
 LDLIBS:= -lpthread
Index: hurd-debian/exec/Makefile
===================================================================
--- hurd-debian.orig/exec/Makefile
+++ hurd-debian/exec/Makefile
@@ -22,7 +22,7 @@ makemode := server
 
 SRCS = exec.c main.c hashexec.c hostarch.c
 OBJS = main.o hostarch.o exec.o hashexec.o \
-       execServer.o exec_startupServer.o exec_experimentalServer.o
+       execServer.o exec_startupServer.o exec_experimentalServer.o processUser.o
 
 target = exec exec.static
 HURDLIBS = trivfs fshelp iohelp ports ihash shouldbeinlibc
Index: hurd-debian/libps/Makefile
===================================================================
--- hurd-debian.orig/libps/Makefile
+++ hurd-debian/libps/Makefile
@@ -29,7 +29,7 @@ installhdrs = ps.h
 installhdrsubdir = .
 
 HURDLIBS=ihash shouldbeinlibc
-OBJS = $(SRCS:.c=.o) msgUser.o termUser.o
+OBJS = $(SRCS:.c=.o) msgUser.o termUser.o processUser.o
 
 msg-MIGUFLAGS = -D'MSG_IMPORTS=waittime 1000;' -DUSERPREFIX=ps_
 term-MIGUFLAGS = -D'TERM_IMPORTS=waittime 1000;' -DUSERPREFIX=ps_
Index: hurd-debian/startup/Makefile
===================================================================
--- hurd-debian.orig/startup/Makefile
+++ hurd-debian/startup/Makefile
@@ -21,7 +21,7 @@ makemode := server
 SRCS = startup.c
 OBJS = $(SRCS:.c=.o) \
        startupServer.o notifyServer.o startup_replyUser.o msgServer.o \
-       startup_notifyUser.o fsysServer.o fsServer.o ioServer.o
+       startup_notifyUser.o fsysServer.o fsServer.o ioServer.o processUser.o
 target = startup
 HURDLIBS = shouldbeinlibc
 
